### PR TITLE
Add sofa-rpc integration for context propagation.

### LINKED ISSUE #1042: Add sofa-rpc integration for context propagation.
## Why you need it?
Add sofa-rpc integration for context propagation.

## How it could be?
- Add integration/sofa-rpc module.
- Add auto active filter for sofa-rpc.
- Add the test case.

### PR DESCRIPTION
**Pull request by ujjboy**
### Ⅰ. Describe what this PR did

- Add integration/sofa-rpc module.
- Add auto active filter for sofa-rpc.
- Add the test case for this.

### Ⅱ. Does this pull request fix one issue?

Fix #1042

### CODE REVIEW

File: bom/pom.xml [hunk 1]
```diff
@@ -91,6 +92,11 @@
                 <artifactId>druid</artifactId>
                 <version>${druid.version}</version>
             </dependency>
+            <dependency>
+                <groupId>com.alipay.sofa</groupId>
+                <artifactId>sofa-rpc-all</artifactId>
+                <version>${sofa.rpc.version}</version>
```
Comment (zhangthen) [inline bom/pom.xml]: set the scope to 'provided' ?

File: integration/sofa-rpc/pom.xml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    <dependencies>
+        <dependency>
+            <groupId>${project.groupId}</groupId>
+            <artifactId>seata-tm</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>com.alipay.sofa</groupId>
+            <artifactId>sofa-rpc-all</artifactId>
+            <scope>provided</scope>
```
Comment (zhangthen) [inline integration/sofa-rpc/pom.xml]: I saw 'provided' here.

Comment (zhangthen) [inline integration/sofa-rpc/pom.xml]: I saw 'provided' here.

File: integration/sofa-rpc/src/main/java/io/seata/integration/sofa/rpc/TransactionContextProviderFilter.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+import com.alipay.sofa.rpc.filter.FilterInvoker;
+import io.seata.core.context.RootContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * TransactionContext on provider side.
+ */
+@Extension(value = "transactionContextProvider")
+@AutoActive(providerSide = true)
```
Comment (slievrly) [inline integration/sofa-rpc/src/main/java/io/seata/integration/sofa/rpc/TransactionContextProviderFilter.java]: Why are consumerSide and providerSide not combined?
@AutoActive(providerSide = true，consumerSide = true)

Comment (ujjboy) [reply integration/sofa-rpc/src/main/java/io/seata/integration/sofa/rpc/TransactionContextProviderFilter.java]: There are some different logic.

**Comment by ujjboy** [comment]
@zhangthen @slievrly   done.

**Pull request merged**